shell/components/fleet/dashboard/ResourcePanel.vue

```diff
@@ -0,0 +1,365 @@
+<script>
+import { markRaw } from 'vue';
+import { mapGetters } from 'vuex';
+import { lcFirst } from '@shell/utils/string';
+import { Chart, registerables } from 'chart.js';
+import { BadgeState } from '@components/BadgeState';
+import FleetUtils from '@shell/utils/fleet';
+
+Chart.register(...registerables);
+
+export default {
+
+  name: 'FleetDashboardResourcePanel',
+
+  emits: ['click:state'],
+
+  components: { BadgeState },
+
+  props: {
+    states: {
+      type:    Object,
+      default: () => ({})
+    },
+
+    workspace: {
+      type:    String,
+      default: ''
+    },
+
+    type: {
+      type:    String,
+      default: ''
+    },
+
+    showChart: {
+      type:    Boolean,
+      default: true
+    },
+
+    selectable: {
+      type:    Boolean,
+      default: true
+    },
+
+    selectedStates: {
+      type:    Object,
+      default: () => ({})
+    },
+  },
+
+  data() {
+    return {
+      chartId:      `${ this.workspace }-${ this.type }`,
+      chart:        null,
+      chartOptions: {
```

Comment:
Consider moving `chartOptions` outside of the data prop. Based on the usage in the `mounted()` hook, it doesn't look like this needs to be made reactive at all. The addition of methods in the data is also a pattern that we want to avoid.